### PR TITLE
Base Notifications!

### PR DESCRIPTION
**Pull request by TempusShift**
Currently doesn't include the DnD time stuff. But now if you run the test_matching and test_expired unit tests you will see posts in the #iftt-spam channel on slack.

### CODE REVIEW

**Comment by TempusShift** [comment]
#14 and #16 #18 domino off of this as well.

**Comment by TempusShift** [comment]
This also Closes #15 & Closes #13.

File: knockserver/notifications.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            if(notification.failed_attempts_threshold >= 0 and device.failure_count % notification.failed_attempts_threshold == 0):
+                send_query(notification.failure_endpoint, user.ifttt_secret)
+
+##
+# Subtracts 5670 minutes (4 hours) to get
+# into eastern time-zone.
+#
+def in_dnd_mode(notification):
+    utc_seconds_today = time.time() % 86400
+    edt_seconds_today = utc_seconds_today - 14400
```
Comment (corbanmailloux) [inline knockserver/notifications.py]: So are we assuming that all users are in Eastern time for now?
I don't think that's an issue with the scope of our project, I just want to make sure that's intended.

File: knockserver/views.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 from flask import Flask, request, Response, render_template, jsonify, json
 
+import requests
```
Comment (corbanmailloux) [inline knockserver/views.py]: Unused.

File: knockserver/notifications.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        for notification in NotificationPreferences.query.filter(NotificationPreferences.id == profile.id).all():
+            
+            if(notification.failed_attempts_threshold >= 0 and device.failure_count % notification.failed_attempts_threshold == 0):
+                send_query(notification.failure_endpoint, user.ifttt_secret)
+
+##
+# Subtracts 5670 minutes (4 hours) to get
+# into eastern time-zone.
+#
+def in_dnd_mode(notification):
```
Comment (corbanmailloux) [inline knockserver/notifications.py]: When is "DND Mode" supposed to be checked? Currently, nothing calls this function. 
I'd expect that it should be checked inside of `send_query` or the other `send_*` functions.

**Review by corbanmailloux — APPROVED** [review]
See my comments for the minor changes. 
I'm interested to see when we have more testing implemented, but this looks good.

File: knockserver/notifications.py (hunk 1 above)
Comment (JakeDarling) [reply knockserver/notifications.py]: Can we throw a comment in here

**Review by JakeDarling — APPROVED** [review]

**Review by ZachCalfin — APPROVED** [review]

**Review by phillip2048 — APPROVED** [review]

**Review by Matthew-Crawford — APPROVED** [review]

**Pull request merged**